### ISSUE TITLE
Cannot allocate memory error on large rows.

### ISSUE THREAD

**Issue by dmnfortytwo** [fluent/fluentd#2093]
Hello!

```
2018-08-06 06:52:55 +0000 [info]: gem 'fluent-plugin-elasticsearch' version '2.4.0'
2018-08-06 06:52:55 +0000 [info]: gem 'fluent-plugin-kafka' version '0.7.2'
2018-08-06 06:52:55 +0000 [info]: gem 'fluent-plugin-rewrite-tag-filter' version '2.0.1'
2018-08-06 06:52:55 +0000 [info]: gem 'fluent-plugin-s3' version '1.1.0'
2018-08-06 06:52:55 +0000 [info]: gem 'fluent-plugin-td' version '1.0.0'
2018-08-06 06:52:55 +0000 [info]: gem 'fluent-plugin-td-monitoring' version '0.2.4'
2018-08-06 06:52:55 +0000 [info]: gem 'fluent-plugin-webhdfs' version '1.2.2'
2018-08-06 06:52:55 +0000 [info]: gem 'fluentd' version '1.2.2'
```

```
<source>
  @type tail
  path /var/log/some_logs/*.log
  tag parsed.dummy.var_log_some_logs
  pos_file /var/lib/td-agent/positions/dummy-var_log_some_logs.db
  read_from_head false
  path_key fname
  <parse>
    @type regexp
    expression ^(?<message>.+)$
  </parse>
</source>
```

We have very long lines in log (>100Kb) and fluent fails on them:

`"<very long line" error="failed to allocate memory: /^(?<message>.+)$/"`

There is no lack of system memory:

```
# free
             total       used       free     shared    buffers     cached
Mem:     429300684  427368296    1932388       9964      58700  349361884
-/+ buffers/cache:   77947712  351352972
Swap:            0          0          0
```

**Comment by dmnfortytwo** [comment]
Well, this happens with more complex regexp's too, I can't replace them with none.

ulimit -m sayes 'unlimited' for td-agent user.

**Comment by fujimotos** [comment]
> ulimit -m sayes 'unlimited' for td-agent user.

I have set up td-agent 3.2 on my machine (Debian 9.2) and tested your
configuration snippet ... and failed to reproduce your issue.

In fact, I can confirm that `in_tail` + `parse_regexp` can handle a file
with very large lines (> 1MB), even on a machine with very limited resource.
(my testing machine has no more than 1GB of RAM).

**Comment by fujimotos** [comment]
> ulimit -m sayes 'unlimited' for td-agent user.

I have set up td-agent 3.2 on my machine (Debian 9.2) to investigate
your issue. I have prepared a file filled with 500KB lines and set up
td-agent using your configuration snippet.

The result was that I cannot reproduce your issue.

In fact, I can confirm that `in_tail` + `parse_regexp` can handle very
large lines (even >1mb), even on a machine with very limited resource.
(my testing machine has no more than 1GB of RAM).

Provided that your system have 300 GB of available memory (including
areas temporarily used for cache) and no resource restriction is posed
on processes, I doubt that 100kb lines were the actual cause of the
memory exhaustion.
